resources/analysisTools/qcPipeline/chrom_diff.r

```diff
@@ -19,8 +20,14 @@ for(i in 1:nrow(data)){
   v[i] <- sum(data[i,])
 }
 
-pdf(opt$outputFile, height = 10, width = 10)
-
-barplot(v, names.arg = rownames(data), col = rainbow(nrow(data)),ylab = "# reads", xlab = "Chromosome", main = plot_title)
+# fix for https://github.com/DKFZ-ODCF/AlignmentAndQCWorkflows/issues/63
+oldWD = getwd()
+setwd(dirname(opt$outputFile))
+tmpFileName=paste0(UUIDgenerate(),".tmp")
```

Comment:
Used like this, the output name has nothing to do with the process that generates it. What about putting the diffChrom/chromDiff information into the name? Think of a situation, where files are **not** sorted into directories.

Reply:
do you mean a situation where opt$outputFile would have no path, just the basename? In how far would this have an impact on the result file (as we still perform the mv aka rename after having created the pdf)?

Reply:
I mean a situation in which files are written outside the diffchrom directory (through changing the filenamepattern), or if the whole script is moved into a new workflow, where file-locations may be reorganized. In that situation it is not clear which process created the file without searching the logs -- for instance if an I/O error occurs during writing the file. Even if we do not put the read-group name into the name (like we decided), just the information from which process the file was created can be helpful for debugging (e.g. to find the relevant I/O error message).